packages/dashboards/src/lib/components/timeseries-widget/chart-presets/timeseries-chart.component.ts

```diff
@@ -152,6 +163,19 @@ export abstract class TimeseriesChartComponent<T = ITimeseriesWidgetSeriesData>
                     this.buildChart();
                 }
 
+                // save original data and transform it
+                this.widgetData.series.forEach((serie) => {
+                    serie.rawData = serie.data;
+                    serie.transformer =
+                        changes.widgetData.previousValue?.series.find(
+                            (prevSerie: ITimeseriesWidgetData) =>
+                                prevSerie.id === serie.id
+                        )?.transformer;
+                    if (serie.transformer && serie.rawData.length > 0) {
+                        serie.data = serie.transformer(serie.rawData);
```

Comment:
same here, could we do the copy only in the case that the transormation will be applied?

````
if (serie.transformer && serie.data.length > 0) {
  serie.rawData = [...serie.data];
  serie.data = serie.transformer(serie.data);
}
````